src/ray/object_manager/object_manager.cc

```diff
@@ -277,8 +277,12 @@ void ObjectManager::HandlePushTaskTimeout(const ObjectID &object_id,
                    << " after waiting for " << config_.push_timeout_ms << " ms.";
   auto iter = unfulfilled_push_requests_.find(object_id);
   RAY_CHECK(iter != unfulfilled_push_requests_.end());
-  size_t num_erased = iter->second.erase(node_id);
-  RAY_CHECK(num_erased == 1);
+  auto node_it = iter->second.find(node_id);
+  RAY_CHECK(node_it != iter->second.end());
+  if (node_it->second) {
+    node_it->second->cancel();
```

Comment:
Given that we only set one handler per timer, and given that `HandlePushTaskTimeout` is only ever reached when the timer expires, what are the expected semantics of cancelling the timer here? It [looks like](https://github.com/boostorg/asio/blob/933e9373d2814c7e823c32cd6c242eb7222906c9/include/boost/asio/basic_deadline_timer.hpp#L335-L343) any other existing handlers would still get invoked with a non-abort error code.

Reply:
The check failure only happens in 2 cases;

1. when the object is added, and then this timer is expired (it seems not possible looking at code)
2. For some unknown reason, the task timeout is scheduled twice. 

I don’t know if 2 is possible, but if it is, this can be the cause of the check failure. This is added to avoid 2 (and this must be the safe change anyway)

Reply:
My expected semantics is this will be no-op

Reply:
I agree with Clark here that this change doesn't really make sense. The timer should already be canceled at this point. Let's make sure to understand the root cause of the problem before we make changes.